# Test RandomForestClassifier:

#### 1. Use the iris.csv dataset

In [1]:
from si.io.csv_file import read_csv

In [2]:
df_path = r"C:\Users\Mariana\si\datasets\iris\iris.csv"
df = read_csv(df_path, sep = ',', features = True, label = True) 

2. Split the data into train and test sets

In [3]:
from si.model_selection.split import train_test_split
from si.models.random_forest_classifier import RandomForestClassifier

In [4]:
train_dataset, test_dataset = train_test_split(df, test_size=0.33, random_state=42)

In [5]:
print(f'(As dimensões dos dados de treino correspondem a:', train_dataset.shape())
print(f'(As dimensões dos dados de teste correspondem a:',test_dataset.shape())

(As dimensões dos dados de treino correspondem a: (101, 4)
(As dimensões dos dados de teste correspondem a: (49, 4)


Os dados de treino apresentam 101 linhas e 4 colunas, enquanto que os dados de teste apresentam 49 linhas e 4 colunas.

3. Create the Random Forest Classifier model

In [6]:
model = RandomForestClassifier()

In [7]:
model

4. Train the model. What is the score of the model on the test set?

In [8]:
model.fit(train_dataset)

In [9]:
model.score(test_dataset)

91.42857142857143

O score de 91.43 indica que o modelo obteve uma precisão de cerca de 91.43% no conjunto de dados de teste. Este valor sugere que o modelo utilizado é bastante bom para a classificação no conjunto de teste.